src/main/java/org/apache/bcel/util/AbstractClassPathRepository.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package org.apache.bcel.util;
+
+import org.apache.bcel.classfile.ClassParser;
+import org.apache.bcel.classfile.JavaClass;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This abstract class provides a logic of a loading {@link JavaClass} objects class names via {@link ClassPath}.
+ *
+ * <p>Subclasses can choose caching strategy of the objects by implementing the abstract methods (e.g., {@link
+ * #storeClass(JavaClass)} and {@link #findClass(String)}).
+ *
+ * @since 6.4.0
+ */
+abstract class AbstractClassPathRepository implements Repository {
+
+    private final ClassPath _path;
+
+    AbstractClassPathRepository(final ClassPath classPath) {
+        _path = classPath;
+    }
+
+    @Override
+    public abstract void storeClass(final JavaClass javaClass);
+
+    @Override
+    public abstract void removeClass(final JavaClass javaClass);
+
+    @Override
+    public abstract JavaClass findClass(final String className);
+
+    @Override
+    public abstract void clear();
+
+    /**
+     * Finds a JavaClass object by name. If it is already in this Repository, the Repository version is returned.
+     * Otherwise, the Repository's classpath is searched for the class (and it is added to the Repository if found).
+     *
+     * @param className
+     *            the name of the class
+     * @return the JavaClass object
+     * @throws ClassNotFoundException
+     *             if the class is not in the Repository, and could not be found on the classpath
+     */
+    @Override
+    public JavaClass loadClass(String className) throws ClassNotFoundException {
+        if ((className == null) || className.isEmpty()) {
```

Comment:
Don't need parens in `(className == null)`

Reply:
Updated.